app/src/main/java/net/gsantner/markor/util/ShortcutUtils.java

```diff
@@ -0,0 +1,131 @@
+/*#######################################################
+ *
+ *   Maintained by Gregor Santner, 2017-
+ *   https://gsantner.net/
+ *
+ *   License of this file: Apache 2.0 (Commercial upon request)
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+#########################################################*/
+package net.gsantner.markor.util;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.ShortcutInfo;
+import android.content.pm.ShortcutManager;
+import android.graphics.drawable.Icon;
+import android.os.Build;
+import android.support.annotation.NonNull;
+
+import net.gsantner.markor.R;
+import net.gsantner.markor.activity.DocumentActivity;
+import net.gsantner.markor.activity.openeditor.OpenEditorQuickNoteActivity;
+import net.gsantner.markor.activity.openeditor.OpenEditorTodoActivity;
+import net.gsantner.markor.format.markdown.MarkdownTextConverter;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Utilities for app shortcuts.
+ * https://developer.android.com/guide/topics/ui/shortcuts
+ */
+public class ShortcutUtils {
+
+    private static final String ID_PREFIX = "shortcut_";
+    private static final String ID_TO_DO = ID_PREFIX + "to_do";
+    private static final String ID_QUICK_NOTE = ID_PREFIX + "quick_note";
+
+    private static final int MAX_RECENT_DOCUMENTS = 2;
+
+    private ShortcutUtils() {
+
+    }
+
+    /**
+     * Update the app shortcuts.
+     * The list will contain a link to to-do, QuickNote and 2 recent documents.
+     *
+     * @param context Context
+     */
+    public static void setShortcuts(@NonNull Context context) {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N_MR1) return;
+
+        ShortcutManager shortcutManager = context.getSystemService(ShortcutManager.class);
+        List<ShortcutInfo> newShortcuts = new ArrayList<>();
+
+        // Create the to-do shortcut
+        Intent intentToDo = new Intent(context, OpenEditorTodoActivity.class);
+        intentToDo.setAction(Intent.ACTION_VIEW);
+
+        ShortcutInfo shortcutToDo = new ShortcutInfo.Builder(context, ID_TO_DO)
+                .setShortLabel(createShortLabel(context.getString(R.string.todo)))
+                .setLongLabel(createLongLabel(context.getString(R.string.todo)))
+                .setIcon(Icon.createWithResource(context, R.mipmap.ic_shortcut_todo))
```

Comment:
Can you please try to use `ic_launcher_linkbox / todo / quicknote` for the static options? This means way less assets :D. If it looks OK, please remove the new quicknote/todo/linkbox pngs

Reply:
It works, but it's not according to the design guidelines. That's why I created new ones.
If the existing resources should be used, they should be upgraded to mipmaps. Those are more fit for launchers.

[Here is a preview](https://i.imgur.com/AYmqLnS.png).

Reply:
Honestly: Prefer the right one, because it's like the launcher :D.
For the normal document I agree that the other should be used